go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go

```diff
@@ -608,3 +608,25 @@ func (jsIPManager *JoinSwitchIPManager) ReleaseJoinLRPIPs(nodeName string) (err
 	}
 	return err
 }
+
+// NewL2SwitchManager initializes a new layer2 logical switch manager,
+// only manage subnet for the one specified switch
+func NewL2SwitchManager() *LogicalSwitchManager {
+	return &LogicalSwitchManager{
+		cache:    make(map[string]logicalSwitchInfo),
+		RWMutex:  sync.RWMutex{},
+		ipamFunc: NewL2IPAMAllocator,
```

Comment:
opening another comment since I cannot 'unresolve' the existing one; we do have a use case for networks without IPAM for flat L2 - and localnet.

Our users coming from RHV (oVirt) / OpenStack / VmWare are used to managing their IP addresses, and we want their user experience to be similar in regards to this.

@cathy-zhou / @trozet / @jcaamano / @girishmg

Reply:
@maiqueb thanks, we can discuss in the next meeting.

though as I understand, to support layer2/localnet network without IPAM, we'd need to introduce additional field in netconf to incidate the type of layer2 network, right? Or shall we just leave "subnets" to be empty and that is all we need?

regardless, supporting no-IPAM network can be done in a later PR. do you see any change needed in the PR to allow that?

Reply:
I thought a bit about this, and we have 3 options to support:
- ovn-k provided IPAM (what this PR currently adds)
- OVN provided IPAM (what @qinqon 's [after](https://github.com/ovn-org/ovn-kubernetes/pull/3350#discussion_r1071165185))
- no IPAM at all

Having an empty subnet would be enough for the last item, but we'd still have to differentiate between the other two (CNI provided IPAM or OVN provided IPAM).

I guess we could revisit this later on, providing a way to optionally configure IPAM.

@qinqon any thoughts ?

Reply:
> I thought a bit about this, and we have 3 options to support:
> - ovn-k provided IPAM (what this PR currently adds)
> - OVN provided IPAM (what @qinqon 's [after](https://github.com/ovn-org/ovn-kubernetes/pull/3350#discussion_r1071165185))
> - no IPAM at all
> 
> Having an empty subnet would be enough for the last item, but we'd still have to differentiate between the other two (CNI provided IPAM or OVN provided IPAM).
> 
> I guess we could revisit this later on, providing a way to optionally configure IPAM.
> 
> @qinqon any thoughts ?

With the introduction of interconnect the ovn IPAM will not be centralized, that could present a problem.

Reply:
It's not really an OVN IPAM, is more like OVN-K IPAM with OVN DHCP, isn't it? From that perspective I don't think IPAM itself is a problem. Are you referring to a problem in regards to distributing /32 routes?

Reply:
> It's not really an OVN IPAM, is more like OVN-K IPAM with OVN DHCP, isn't it? From that perspective I don't think IPAM itself is a problem. Are you referring to a problem in regards to distributing /32 routes?

Well originally we were using "dynamic" at the LSP address so it was using ovn IPAM, but after find out the interconnect thing we switched to ovn-k setting the address, at the end is the same for us.

So yes, ovn-k ipam + DHCP, if in the future ovn implement distributed ipam across zones maybe ovn-k can delegate on that.